cpp/src/arrow/compute/kernels/aggregate_basic_internal.h

```diff
@@ -220,16 +220,24 @@ struct MinMaxState<ArrowType, SimdLevel, enable_if_boolean<ArrowType>> {
     this->max = this->max || value;
   }
 
+  Status GetValues(const std::shared_ptr<DataType>&,
+                   std::vector<std::shared_ptr<Scalar>>* values) {
+    values->emplace_back(std::make_shared<BooleanScalar>(min));
+    values->emplace_back(std::make_shared<BooleanScalar>(max));
+    return Status::OK();
+  }
+
   T min = true;
   T max = false;
   bool has_nulls = false;
   bool has_values = false;
 };
 
 template <typename ArrowType, SimdLevel::type SimdLevel>
-struct MinMaxState<ArrowType, SimdLevel, enable_if_integer<ArrowType>> {
+struct MinMaxState<ArrowType, SimdLevel, enable_if_physical_integer<ArrowType>> {
```

Comment:
It seems this will generate different template specializations for types with the same underlying `c_type`. Is it desirable?
(note that `GetValues` is called with the output type independently from the `ArrowType` parameter)

Reply:
I changed this and the one below to use PhysicalType. I also replaced GetValues with MakeScalar (since now the static type doesn't match the runtime type).